clients/drcachesim/tests/burst_replace.cpp

```diff
@@ -182,23 +182,22 @@ post_process()
     assert(mem_res == DRMEMTRACE_SUCCESS);
     std::string output_trace(std::string(output_path) + std::string(DIRSEP) + ".." +
                              std::string(DIRSEP) + TRACE_FILENAME);
+    void *dr_context = dr_standalone_init();
     {
         /* First, test just the module parsing w/o writing a final trace, in a separate
          * scope to delete the drmodtrack state afterward.
          */
         raw2trace_directory_t dir(output_path, output_trace);
-        raw2trace_t raw2trace(dir.modfile_bytes, dir.thread_files, NULL, NULL);
-        std::string error =
-            raw2trace.handle_custom_data(parse_cb, process_cb, MAGIC_VALUE, free_cb);
-        assert(error.empty());
-        error = raw2trace.do_module_parsing();
-        assert(error.empty());
+        module_mapper_t module_mapper(dir.modfile_bytes, parse_cb, process_cb,
+                                      MAGIC_VALUE, free_cb);
+        assert(module_mapper.get_last_error().empty());
     }
     /* Now write a final trace to a location that the drcachesim -indir step
      * run by the outer test harness will find (TRACE_FILENAME).
      */
     raw2trace_directory_t dir(output_path, output_trace);
-    raw2trace_t raw2trace(dir.modfile_bytes, dir.thread_files, &dir.out_file, NULL, 0);
+    raw2trace_t raw2trace(dir.modfile_bytes, dir.thread_files, &dir.out_file, dr_context,
```

Comment:
because deeper down in the call chain, a dr context is assumed to be created. Before, raw2trace_t was constructing one if none was provided. raw2trace_t also needed one for other reasons (e.g. its hashtable).

We could do a similar thing here, where users are required to pass a drcontext, albeit it wouldn't be used by module_tracer_t, just used as a signal of whether one were already created; or (current status) document that one should be present.